File: Binary/binary.py
import sys

def max(a,b):
    return a if a>b else b

n = int(input().strip())

max_num = 0
# Change below line to pass all test cases.
count = n

while n:
    # Change below line to pass all test cases.
    while n&2:
        count += 1
        n>>=1
    max_num = max(count, max_num)
    if not n&1:
        count = 0
        # Change below line to pass all test cases.
        n>>=2

print("Maximum consecutive 1 in binary:", max_num)

File: Binary/binary.cpp
int main(){
    int n;
    cin >> n;
    int max = 0;
    // Change below line to pass all test cases.
    int count = n;

    // Change below line to pass all test cases.
    while(n>=0)
    {
        if(n%2==1)
            count++;
        else
            count = 0;
        if(max<count)
            // Change below line to pass all test cases.
            max = count*2;
        n=n/2;
    }
    cout<<"Maximum consecutive 1 in binary: " <<max;
    return 0;
}